nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/authentication/jwt/JwtService.java

```diff
@@ -155,6 +176,7 @@ public String generateSignedToken(String identity, String preferredUsername, Str
                     .audience().add(audience).and()
                     .claim(USERNAME_CLAIM, preferredUsername)
                     .claim(KEY_ID_CLAIM, key.getId())
+                    .claim(GROUPS_CLAIM, groups)
```

Comment:
This `groups` claim value can be null in some cases, such as authentication with LDAP. Instead of adding a `groups` claim with a `null` value, either an empty list should be set, or the claim should be omitted. As this is the JWT that NiFi Registry generates, it seems better to set an empty list when the input `groups` parameter is null.